internal/web/applications_controller.go

```diff
@@ -0,0 +1,76 @@
+package web
+
+import (
+	"html/template"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/suse/carrier/internal/filesystem"
+)
+
+type ApplicationsController struct {
+}
+
+func (hc ApplicationsController) Index(w http.ResponseWriter, r *http.Request) {
+	Render([]string{"main_layout", "icons", "applications_index"}, w, r, map[string]interface{}{"serverUrl": r.Host})
+}
+
+// Render renders the given templates using the provided data and writes the result
+// to the provided ResponseWriter.
+func Render(templates []string, w http.ResponseWriter, r *http.Request, data map[string]interface{}) {
+	var viewsDir http.FileSystem
+	if os.Getenv("LOCAL_FILESYSTEM") == "true" {
+		viewsDir = http.Dir(path.Join(".", "embedded-web-files", "views"))
+	} else {
+		viewsDir = filesystem.Views()
+	}
+
+	var err error
+	tmpl := template.New("page_template")
+	tmpl = tmpl.Delims("[[", "]]")
+	for _, template := range templates {
+		tmplFile, err := viewsDir.Open("/" + template + ".html")
+		if err != nil {
+			if handleError(w, err, 500) {
+				return
+			}
+		}
+		tmplContent, err := ioutil.ReadAll(tmplFile)
+		if err != nil {
+			if handleError(w, err, 500) {
+				return
+			}
+		}
+
+		tmpl, err = tmpl.Parse(string(tmplContent))
+		if err != nil {
+			if handleError(w, err, 500) {
+				return
+			}
+		}
+	}
+
+	if handleError(w, err, 500) {
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+
+	// Inject the request in the data. It can be used in the templates to find the
+	// current url
```

Comment:
I cannot find where the current_url is being used in the templates.

Reply:
https://github.com/SUSE/carrier/pull/235/files#diff-ecc48de753581438ba869ceff1df561c7fcf30b3baf38d6e19b20a658e11f6a8R73

I thought about Rails and how the request information is available inside the views. We could simply inject just the r.URL.Path but I thought other attributes of the request may be useful in the future.